### PR TITLE
Update slither-analyzier & crytic-compiler & echidna

### PR DESCRIPTION
**Pull request by hellwolf**
###### Description of changes

*  python311Packages.slither-analyzer: 0.9.2 -> 0.9.6
*  python311Packages.crytic-compile: 0.3.0 -> 0.3.3
*  echidna: 2.0.5 -> 2.2.1
*  add @hellwolf to the maintainer list too all these packages

This requires these to be merged:

- https://github.com/NixOS/nixpkgs/pull/230065
- https://github.com/NixOS/nixpkgs/pull/240007

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hellwolf** [comment]
@ofborg build slither-analyzer crytic-compile

**Comment by hellwolf** [comment]
https://github.com/NixOS/nixpkgs/pull/230065 is now merged.

**Review by fabaff — COMMENTED** [review]
Commit messages should follow `CONTRIBUTING.md` .

If you decide not to use `nix-update` then you would have to add the changelogs manually to the commit messages.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 230067` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.crytic-compile</li>
    <li>python310Packages.crytic-compile.dist</li>
    <li>slither-analyzer (python310Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python310Packages.slither-analyzer.dist)</li>
    <li>python311Packages.crytic-compile</li>
    <li>python311Packages.crytic-compile.dist</li>
    <li>python311Packages.slither-analyzer</li>
    <li>python311Packages.slither-analyzer.dist</li>
  </ul>
</details>

**Comment by hellwolf** [comment]
> Commit messages should follow `CONTRIBUTING.md` .
> 
> If you decide not to use `nix-update` then you would have to add the changelogs manually to the commit messages.

Aha, okay, I will need to learn nix-update first then...

I updated the commit title at least, for now.

File: pkgs/tools/security/echidna/default.nix [hunk 1]
```diff
@@ -1,36 +1,31 @@
 { lib
```
Comment (SuperSandro2000) [inline pkgs/tools/security/echidna/default.nix:1]: Isn't this mostly auto generated?

Comment (hellwolf) [reply pkgs/tools/security/echidna/default.nix:1]: good question, it doesn't seem to me so, I will check with the maintainer.

**Comment by hellwolf** [comment]
I upgraded to echidna 2.2.0.

**Comment by hellwolf** [comment]
@ofborg build slither-analyzer echidna

**Comment by hellwolf** [comment]
I overrode hevm version, and echidna-2.2.0 compiles too now.

**Comment by hellwolf** [comment]
All platform builds pass.

**Comment by fabaff** [comment]
Commit messages for Python module should be prefix with `python311Packages.`.

**Comment by hellwolf** [comment]
`$ nixpkgs-review pr --print-result 230067`:

Result of `nixpkgs-review pr 230067` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>echidna</li>
    <li>python310Packages.crytic-compile</li>
    <li>python310Packages.crytic-compile.dist</li>
    <li>slither-analyzer (python310Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python310Packages.slither-analyzer.dist)</li>
    <li>python311Packages.crytic-compile</li>
    <li>python311Packages.crytic-compile.dist</li>
    <li>python311Packages.slither-analyzer</li>
    <li>python311Packages.slither-analyzer.dist</li>
  </ul>
</details>

**Comment by hellwolf** [comment]
FOLKS, it's ready!

@fabaff @arcz etc.

File: pkgs/development/python-modules/crytic-compile/default.nix [hunk 2]
```diff
@@ -18,27 +19,31 @@ buildPythonPackage rec {
...
 
   # Test require network access
   doCheck = false;
 
+  # required for import check to work
+  # PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
+  env.HOME = "/tmp";
   pythonImportsCheck = [
-    "crytic_compile"
+   "crytic_compile"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/crytic-compile/default.nix:39]:
```suggestion
    "crytic_compile"
```
this was correct

File: pkgs/development/python-modules/crytic-compile/default.nix [hunk 3]
```diff
@@ -18,27 +19,31 @@ buildPythonPackage rec {
...
     setuptools
+    solc-select
   ];
 
   # Test require network access
   doCheck = false;
 
+  # required for import check to work
+  # PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
+  env.HOME = "/tmp";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/crytic-compile/default.nix:37]: normally we do in preCheck:

```
export HOME=$(mktemp -d)
```

File: pkgs/development/python-modules/crytic-compile/default.nix [hunk 4]
```diff
@@ -18,27 +19,31 @@ buildPythonPackage rec {
...
+   "crytic_compile"
   ];
 
   meta = with lib; {
     description = "Abstraction layer for smart contract build systems";
     homepage = "https://github.com/crytic/crytic-compile";
     changelog = "https://github.com/crytic/crytic-compile/releases/tag/${version}";
     license = licenses.agpl3Plus;
-    maintainers = with maintainers; [ SuperSandro2000 arturcygan ];
+    maintainers = with maintainers; [ SuperSandro2000 arturcygan hellwolf ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/crytic-compile/default.nix:47]:
```suggestion
    maintainers = with maintainers; [ arturcygan hellwolf ];
```

File: pkgs/tools/security/echidna/default.nix [hunk 5]
```diff
@@ -1,38 +1,60 @@
...
   pname = "echidna";
-  version = "2.0.5";
+  version = "2.2.0";
 
   src = fetchFromGitHub {
     owner = "crytic";
     repo = "echidna";
-    rev = "v${version}";
-    sha256 = "sha256-8bChe+qA4DowfuwsR5wLckb56fXi102g8vL2gAH/kYE=";
+    rev = "v2.2.0";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/echidna/default.nix:30]:
```suggestion
    rev = "v${version}";
```

**Comment by hellwolf** [comment]
@SuperSandro2000 thanks, I am waiting for @arcz for a new package he told me about before merging this.

File: pkgs/development/python-modules/crytic-compile/default.nix (hunk 3 above)
Comment (hellwolf) [reply pkgs/development/python-modules/crytic-compile/default.nix:37]: I tried either 'preCheck' or 'preDist', they had no effect. I am a bit confused.

Comment (hellwolf) [reply pkgs/development/python-modules/crytic-compile/default.nix:37]: see: pkgs/development/interpreters/python/hooks/python-imports-check-hook.sh

```
# Setup hook for checking whether Python imports succeed
echo "Sourcing python-imports-check-hook.sh"

pythonImportsCheckPhase () {
    echo "Executing pythonImportsCheckPhase"

    if [ -n "$pythonImportsCheck" ]; then
        echo "Check whether the following modules can be imported: $pythonImportsCheck"
        export PYTHONPATH="$out/@pythonSitePackages@:$PYTHONPATH"
        ( cd $out && eval "@pythonCheckInterpreter@ -c 'import os; import importlib; list(map(lambda mod: importlib.import_module(mod), os.environ[\"pythonImportsCheck\"].split()))'" )
    fi
}

if [ -z "${dontUsePythonImportsCheck-}" ]; then
    echo "Using pythonImportsCheckPhase"
    preDistPhases+=" pythonImportsCheckPhase"
fi
```

I honestly don't know how to fix it for now.

**Comment by hellwolf** [comment]
Result of `nixpkgs-review pr 230067` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>echidna</li>
    <li>python310Packages.crytic-compile</li>
    <li>python310Packages.crytic-compile.dist</li>
    <li>slither-analyzer (python310Packages.slither-analyzer)</li>
    <li>slither-analyzer.dist (python310Packages.slither-analyzer.dist)</li>
    <li>python311Packages.crytic-compile</li>
    <li>python311Packages.crytic-compile.dist</li>
    <li>python311Packages.slither-analyzer</li>
    <li>python311Packages.slither-analyzer.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/crytic-compile/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/crytic-compile/default.nix:37]: There should be something like preDist, too but yeah, not so important and also slightly annoying that upstream is doing more than just importing when importing

Comment (hellwolf) [reply pkgs/development/python-modules/crytic-compile/default.nix:37]: > slightly annoying that upstream is doing more than just importing when importing

Oh, that reminds me that I wanted to create a ticket in upstream pointing this out.

**Pull request merged**